btc/address.go

```diff
@@ -0,0 +1,36 @@
+package btc
+
+import (
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/chaincfg"
+)
+
+type AddressType string
+
+var (
+	AddressP2PK    AddressType = "P2PK"
+	AddressP2PKH   AddressType = "P2PKH"
+	AddressP2SH    AddressType = "P2SH"
+	AddressP2WPKH  AddressType = "P2WPKH"
+	AddressSegwit  AddressType = "Bech32"
```

Comment:
AddressSegwit is not being used anywhere and p2wpkh is already a segwit address. Any reason to keep it ?

Reply:
AddressSegwit is mostly for future proof and currently is not used anywhere. The whole address file is still experimental. And yeah p2wpkh is a special type address of segwit. Since it is commonly used in a lot places, so it's worth to have a separate type for it, so we know how to how to estimate fees/sign sigature/decode address from key.